### PR TITLE
Improved EarlyStopping.patience documentation

### LINKED ISSUE #6254: Improve documentation for EarlyStopping patience parameter
## 📚 Documentation

[EarlyStopping API docs](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.early_stopping.html#pytorch_lightning.callbacks.early_stopping.EarlyStopping) currently reads:

```
patience (int) – number of validation epochs with no improvement after which training will be stopped. Default: 3.
```

However, this is quite confusing because 'validation epochs' is not really a term used much in the documentation, and this leads the user to believe this is based upon training epochs. 

I would propose something like the following:

```
patience (int) – number of validation checks with no improvement after which training will be stopped. Default: 3. For example, if `check_val_every_n_epoch=4`, early stopping will take at least 12 training epochs to occur.
```

Related: https://github.com/PyTorchLightning/pytorch-lightning/discussions/6253

### PR DESCRIPTION
**Pull request by turian**
## What does this PR do?

The terminology "validation epochs" is a bit confusing, and idiomatic as it is not really used much elsewhere in the documentation. The documentation has been updated to say "validation checks" and explain more clearly how EarlyStopping patience works.

Fixes #6254 

## Before submitting
- [Y] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [Y] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [Y] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [Y] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [Y] Is this pull request ready for review? (if not, please submit in draft mode)
 - [Y] Check that all items from **Before submitting** are resolved
 - [Y] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -38,8 +38,18 @@ class EarlyStopping(Callback):
...
+            configuration, one validation check happens after every
+            training epoch. However, the frequency of validation
+            can be modified by setting various parameters on the
+            Trainer, for example check_val_every_n_epoch and
+            val_check_interval. It must be noted that the patience
+            parameter counts the number of validation checks with
+            no improvement, and not the number of training epochs.
+            Therefore, with parameters check_val_every_n_epoch=10
+            and patience=3, the trainer will perform at least 40
+            training epochs before being stopped. Default: ``3``.
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py:52]: pls use a full 120 line length...

Comment (turian) [reply pytorch_lightning/callbacks/early_stopping.py:52]: All the other comments in this file are at the column length I used. See here:

image

However, I will adjust to 120 and resubmit

Comment (turian) [reply pytorch_lightning/callbacks/early_stopping.py]: @Borda Please see revision

Comment (Borda) [reply pytorch_lightning/callbacks/early_stopping.py]: I see your point, to be clear not all in the codebase is ideal, but as something suboptimal was used in past does not mean we have to keep using it... does it make sense?

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by kaushikb11 — APPROVED** [review]
